cmd/addon_cert-manager.go

```diff
@@ -0,0 +1,27 @@
+package cmd
+
+import "log"
+
+type CertmanagerAddon struct {
+	masterNode *Node
+}
+
+func NewCertmanagerAddon(cluster Cluster) ClusterAddon {
+	masterNode, err := cluster.GetMasterNode()
+	FatalOnError(err)
+	return CertmanagerAddon{masterNode: masterNode}
+}
+
+func (addon CertmanagerAddon) Install(args ... string) {
+	node := *addon.masterNode
+	_, err := runCmd(node, "helm install --name cert-manager --namespace kube-system --set ingressShim.extraArgs='{--default-issuer-name=letsencrypt-prod,--default-issuer-kind=ClusterIssuer}' stable/cert-manager")
```

Comment:
`extraArgs='{--default-issuer-name=letsencrypt-prod,--default-issuer-kind=ClusterIssuer}'`

If you read carefully the doc, it says that this argument is needed if you want backward compatibility with `kube-lego`. (i didn't read the code to be honest)

I did test without, and it works!
https://www.youtube.com/watch?v=BXHyvPdhEFQ
(Look in the description, there are all the params used.)

Reply:
I think these are sane defaults. The user is always free to override them. When adding a ClusterIssues like you posted below and name it accordingly "letsencrypt-prod" it works just fine.